addons/bus/models/bus_presence.py

```diff
@@ -51,5 +67,3 @@ def update(self, inactivity_period):
             # Hide transaction serialization errors, which can be ignored, the presence update is not essential
             with tools.mute_logger('odoo.sql_db'):
                 presence.write(values)
```

Comment:
We could even go one step further, and merge the "create" and "write" into a single "upsert" query (INSERT ... ON CONFLICT ...).  Of course this may only be possible in later versions of Odoo, but it is worth it.  I don't remember how SKIP LOCKED combines with "upsert" queries, though.

Reply:
> For the record, we never noticed any significant disturbance caused by presence update conflicts on our production systems. Probably there's something specific happening for that customer, unusually high duplication of presence updates. It would be well advised to investigate that problem on their side, because the PR only mitigates the impact - there will still be useless presence updates wasting CPU cycles. It could be that a single user is causing all of this (easy to verify in the PG logs)


That customer uses odoo.sh and problem happens rarely and randomly.
I checked Odoo logs and found that 140 request/sec come in some periods from single IP address. But

* they have 60 users
* the IP is their office IP, so it's not possible to identify user

Reply:
>     * the IP is their office IP, so it's not possible to identify user

That's why I mentioned PG logs in the first place ;) The failed UPDATE queries will be logged, including the bus_presence ID that was being updated. This should link to a user, shouldn't it?

Reply:
It's v12 and logs muting is working :D 
That's why https://github.com/odoo/odoo/pull/57067 is for v13

Reply:
If error logs are potentially useful, shall I remove muting? :thinking: 
cc @guewen

Reply:
> It's v12 and logs muting is working :D
> That's why #57067 is for v13

**PG** logs, not Odoo logs!!!

Reply:
PG acronym is clear enough, but I never used postgres logs, that's why I'm confused. And apparently neither saas nor odoo.sh provide access to it.